### PR TITLE
Deployment tooling update - bin/jenkins

### PR DESCRIPTION
**Pull request by indigobravo**
The bin/jenkins script has been updated to handle multi-region deployments.

---

### CODE REVIEW

File: bin/jenkins [hunk 1]
```diff
@@ -14,6 +14,8 @@
 # name of the environment in Elastic Beanstalk will be prefixed with the name of
 # the application.
 : ${ENV:=}
+# The target AWS region
```
Comment (indigobravo) [inline bin/jenkins:17]: I have copied the format of the original script here...

If I am understanding this correct, the `:` basically does nothing, and sets `REGION` to a null value if it does not exist.

File: bin/jenkins [hunk 2]
```diff
@@ -33,24 +35,28 @@ if [ -z "$ENV" ]; then
     abort "cannot proceed unless \$ENV is specified"
 fi
 
+if [ -z "$REGION" ]; then
```
Comment (indigobravo) [inline bin/jenkins:38]: Check if the `REGION` variable is null and exit.

File: bin/jenkins [hunk 3]
```diff
@@ -33,24 +35,28 @@ if [ -z "$ENV" ]; then
...
+fi
+
 if [ "$TYPE" = "deploy" ]; then
   if [ -z "$APP_DOCKER_VERSION" ]; then
     abort "cannot proceed with deploy unless \$APP_DOCKER_VERSION is specified"
   fi
 
   version_label=$(eb-release "$APP" "$APP_DOCKER_VERSION" "$ENV")
-  eb-deploy "$APP" "$ENV" "$version_label"
+  eb-deploy "$APP" "$ENV" "$version_label" "$REGION"
```
Comment (indigobravo) [inline bin/jenkins:48]: Pass `REGION` to `eb-deploy`

File: bin/jenkins [hunk 4]
```diff
@@ -33,24 +35,28 @@ if [ -z "$ENV" ]; then
...
 
 elif [ "$TYPE" = "redeploy" ]; then
   if [ -n "$APP_DOCKER_VERSION" ]; then
     abort "do not specify \$APP_DOCKER_VERSION for a redeploy"
   fi
 
   version_label=$(eb-env-version "$APP" "$ENV")
-  eb-deploy "$APP" "$ENV" "$version_label"
+  eb-deploy "$APP" "$ENV" "$version_label" "$REGION"
```
Comment (indigobravo) [inline bin/jenkins:56]: Pass `$REGION` to `eb-deploy`

File: bin/jenkins [hunk 5]
```diff
@@ -33,24 +35,28 @@ if [ -z "$ENV" ]; then
...
 
 elif [ "$TYPE" = "sync-env" ]; then
-  eb-env-sync "$APP" "$ENV"
+  eb-env-sync "$APP" "$ENV" "$REGION"
```
Comment (indigobravo) [inline bin/jenkins:59]: Pass `$REGION` to `eb-env-sync`

File: bin/jenkins (hunk 1 above)
Comment (robertknight) [reply bin/jenkins:17]: That matches my understanding. `:` is a no-op command that allows `${VAR:=val}` to be executed. I guess this coding pattern is to allow known variables to be "registered" so they don't trigger unknown variable errors if not set.

**Review by robertknight — APPROVED** [review]
LGTM

Edit: I realized there is a hazard here where the aws commands could end up operating against a different region than what is specified via `REGION`. We need to prevent that somehow.

File: bin/jenkins [hunk 6]
```diff
@@ -14,11 +14,15 @@
...
 # the application.
 : ${ENV:=}
+# The target AWS region
+: ${REGION:=}
 
 PATH="$(dirname "$0"):${PATH}"
 
 set -eu
 
+export AWS_DEFAULT_REGION="$REGION"
```
Comment (indigobravo) [inline bin/jenkins:24]: Purposely setting the `AWS_DEFAULT_REGION` environment variable before invoking any commands.

File: bin/jenkins [hunk 7]
```diff
@@ -33,26 +37,30 @@ if [ -z "$ENV" ]; then
...
+fi
+
 if [ "$TYPE" = "deploy" ]; then
   if [ -z "$APP_DOCKER_VERSION" ]; then
     abort "cannot proceed with deploy unless \$APP_DOCKER_VERSION is specified"
   fi
 
-  version_label=$(eb-release "$APP" "$APP_DOCKER_VERSION" "$ENV")
-  eb-deploy "$APP" "$ENV" "$version_label"
+  version_label=$(eb-release "$APP" "$APP_DOCKER_VERSION" "$ENV" "$REGION")
```
Comment (indigobravo) [inline bin/jenkins:49]: Pass `$REGION` to `eb-release`

File: bin/jenkins [hunk 8]
```diff
@@ -33,26 +37,30 @@ if [ -z "$ENV" ]; then
...
 
 elif [ "$TYPE" = "sync-env" ]; then
-  eb-env-sync "$APP" "$ENV"
+  eb-env-sync "$APP" "$ENV" "$REGION"
 else
   abort "Don't know how to handle deployment type \"$TYPE\""
 fi
 
-eb-env-wait "$APP" "$ENV"
+eb-env-wait "$APP" "$ENV "$REGION""
```
Comment (indigobravo) [inline bin/jenkins:66]: Pass `$REGION` to `eb-env-wait`

File: bin/jenkins (hunk 1 above)
Comment (indigobravo) [reply bin/jenkins:17]: Thanks for the clarification.

File: bin/jenkins [hunk 9]
```diff
@@ -33,24 +37,28 @@ if [ -z "$ENV" ]; then
...
+  eb-deploy "$APP" "$ENV" "$version_label" "$REGION"
 
 elif [ "$TYPE" = "redeploy" ]; then
   if [ -n "$APP_DOCKER_VERSION" ]; then
     abort "do not specify \$APP_DOCKER_VERSION for a redeploy"
   fi
 
-  version_label=$(eb-env-version "$APP" "$ENV")
-  eb-deploy "$APP" "$ENV" "$version_label"
+  version_label=$(eb-env-version "$APP" "$ENV" "$REGION")
```
Comment (indigobravo) [inline bin/jenkins:57]: Passing `$REGION` to the `eb-env-version` script

File: bin/jenkins [hunk 10]
```diff
@@ -33,26 +37,30 @@ if [ -z "$ENV" ]; then
...
 
 elif [ "$TYPE" = "sync-env" ]; then
-  eb-env-sync "$APP" "$ENV"
+  eb-env-sync "$APP" "$ENV" "$REGION"
 else
   abort "Don't know how to handle deployment type \"$TYPE\""
 fi
 
-eb-env-wait "$APP" "$ENV"
+eb-env-wait "$APP" "$ENV" "$REGION"
```
Comment (indigobravo) [inline bin/jenkins:66]: Pass `$REGION` to `eb-env-wait`

**Pull request merged**